analysis/dataflow/state.go

```diff
@@ -406,6 +406,10 @@ func (s *AnalyzerState) PopulateBoundingInformation(verbose bool) error {
 // To compute reachable functions without main or init, use the CallGraphReachable function with
 // the appropriate callgraph information.
 func (s *AnalyzerState) ReachableFunctions() map[*ssa.Function]bool {
+	if len(s.reachableFunctions) > 0 {
```

Comment:
I think this is the cause of the failing test. The state will recompute reachability based on the "best available callgraph" which will vary depending on whether pointer information has been computed. Checking whether this has already computed needs to be a little more involved (we can store an enum telling which callgraph computation has been used, and recompute if new information is available).

Reply:
I'd remove this optimization in this PR and we can work on it again later.